app/src/main/java/mozilla/lockbox/presenter/AuthPresenter.kt

```diff
@@ -52,33 +55,38 @@ class AuthPresenter(
 
         lockedStore.onAuthentication
             .subscribe {
-                if (it is FingerprintAuthAction.OnAuthentication) {
-                    when (it.authCallback) {
-                        is FingerprintAuthCallback.OnAuth -> unlock()
-                        is FingerprintAuthCallback.OnError -> view.unlockFallback()
+                when (it) {
+                    is FingerprintAuthAction.OnAuthentication -> {
+                        when (it.authCallback) {
+                            is FingerprintAuthCallback.OnAuth -> unlock()
+                            is FingerprintAuthCallback.OnError -> unlockFallback()
+                        }
                     }
+                    is FingerprintAuthAction.OnCancel -> finishResponse()
                 }
             }
             .addTo(compositeDisposable)
 
         view.unlockConfirmed
-            .filter { it }
             .subscribe {
-                unlock()
+                if (it) {
+                    unlock()
+                } else {
+                    finishResponse()
+                }
             }
             .addTo(compositeDisposable)
 
         dataStore.state
             .filter { it == DataStore.State.Unlocked }
-            .switchMap { responseBuilder.asyncFilter(pslSupport, dataStore.list) }
+            // switch to the latest non-empty list of ServerPasswords
+            // TODO: deal with the list really being empty ...
```

Comment:
I filed #464 to address this.  It's only encountered if the user really truly has an empty list, but it can leave our Autofill service in limbo ...

Reply:
good callout, let's make sure that gets uplifted in sprint planning tomorrow.